Generation/src/components/PileUpEdmFileReaderTool.cpp

```diff
@@ -0,0 +1,81 @@
+
+#include "PileUpEdmFileReaderTool.h"
+#include "datamodel/MCParticleCollection.h"
+#include "datamodel/GenVertexCollection.h"
+
+#include "GaudiKernel/IIncidentSvc.h"
+#include "GaudiKernel/Incident.h"
+#include "GaudiKernel/IEventProcessor.h"
+
+#include "GaudiKernel/DeclareFactoryEntries.h"
+
+#include "podio/EventStore.h"
+#include "podio/ROOTReader.h"
+
+#include "HepMC/GenEvent.h"
+
+DECLARE_TOOL_FACTORY(PileUpEdmFileReader)
+
+PileUpEdmFileReader::PileUpEdmFileReader(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface<IHepMCProviderTool>(this);
+  declareProperty("filename", m_filename = "input.root", "EDM file which contains the particle collections");
+  declareProperty("particleCollectionName", m_particleCollectionName = "allGenParticles");
+  declareProperty("vertexCollectionName", m_vertexCollectionName = "allGenVertices");
+}
+
+PileUpEdmFileReader::~PileUpEdmFileReader() { ; }
+
+StatusCode PileUpEdmFileReader::initialize() {
+  m_store = podio::EventStore();
+  m_reader = podio::ROOTReader();
```

Comment:
I don't like this usage of the reader and event store since it circumvents all the handles and related functionality of Gaudi. I think this needs some further discussion (also with @hegner).